firmware/application/ui_navigation.cpp

```diff
@@ -555,6 +560,76 @@ void SystemStatusView::rtc_battery_workaround() {
     }
 }
 
+void SystemStatusView::new_sdcard_structure_checker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path resources_dir = u"SYS";
+    std::vector<std::filesystem::path> directories = scan_root_directories(root_dir);
+
+    auto scan_result = std::find(directories.begin(), directories.end(), resources_dir);
+
+    auto sd_status = sd_card::status();
+
+    if (
+        nav_.is_valid() &&
+        (scan_result == directories.end()) &&
+        (sd_status == sd_card::Status::Mounted)) {
+        nav_.display_modal(
+            "Warning",
+            "        WARNING!!!\n"
+            "Sdcard content changed\n"
+            "in this update,\n"
+            "Press YES to auto move.\n"
+            "After finished,\n"
+            "you should move your own fil-\n"
+            "es back into original folder,\n"
+            "which is now only for users.\n"
+            "You can also choose NO,\n"
+            "and then manually merge\n"
+            "following the Wiki.\n"
+            "Details:\n"
+            "Mayhem wiki - Updating",
+            COMPACTYESNO,
+            [this](bool choice) {
+                if (choice) {
+                    new_sdcard_structure_worker();
+                }
+            });
+    }
+}
+
+void SystemStatusView::new_sdcard_structure_worker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path system_dir = u"SYS";
+
+    // TODO: Use blacklist or whitelist?
+    std::vector<std::filesystem::path> ignore_dirs = {u"FIRMWARE",
+                                                      u"APPS",
+                                                      u"SYS",
+                                                      u"SETTINGS",
```

Comment:
Should we use the path strings defined in file_path.cpp here too?

Reply:
I guess you didn't like this idea since you clicked "resolved" with no comment.

Reply:
No sorry it’s not the case. 
I can move them into path file for sure if you insist. But I just don’t think it can save much resources (maybe can’t save at all), because it’s only been used once here.